### ISSUE TITLE
Medical antonyms

### ISSUE THREAD

**Issue by kenlkehl** [ncbi-nlp/BioSentVec#17]
The following is not a bug in your code. Rather, wondering if any have thoughts on the following. 
I'm working on some NLP tasks in oncology.
Have found that randomly initialized word/sentence embeddings tend to work better than any pretrained embeddings for ultimately classifying, say, improving vs worsening cancer.
Had an intuition that this might be because key words for telling the two apart tend to be embedded similarly.

In trying BioSentVec, this seems to be borne out, eg:
progression = model.embed_sentence(preprocess_sentence("Increase in size of tumor"))
response = model.embed_sentence(preprocess_sentence("Decrease in size of tumor"))
1 - distance.cosine(progression, response)

Yields: 0.94.  Opposite meanings are embedded similarly, which explains why building classifiers on these embeddings does not work well.

Any methods for addressing this in the transfer learning setting that you're aware of? I have not found any---

**Comment by qingyu-qc** [comment]
Hi,

Thanks for your question. Indeed, as you mentioned, word/sentence embeddings are not designed for tackling antonyms. Similar observations are also seen in BERT https://medium.com/thelocalminima/are-bert-features-interbertible-250a91eb9dc and https://github.com/google-research/bert/issues/507. To my knowledge, one solution is to directly modify the loss of the embeddings during training (http://tcci.ccf.org.cn/conference/2018/papers/141.pdf). Another solution is to do retrofitting, i.e., tune the embedding based on specific datasets.

I close this issue for now but do let us know if you have found promising solutions. 

Thanks.
